csclient/csclient.go

```diff
@@ -655,6 +655,7 @@ func (c *Client) Publish(id *charm.URL, channels []params.Channel, resources map
 // It must be closed after use.
 type ResourceData struct {
 	io.ReadCloser
+	Size int64
```

Comment:
I wonder if this should be a *int64 so that one can tell the difference between a server that just doesn't tell you and a 0 length resource?

Reply:
It'll be -1 if it's not there, like http.Response. It's part of the contract that the charmstore will always tell us. I guess I could document the -1 convention, or just fail if there's no content length in the response.

Reply:
Sorry I didn't manage to spot how it was working, carry on.